Test failures with Python 3.13

Hi!

Python 3.13 has changed the message associated with HTTP code 422 from 'Unprocessable Entity' to 'Unprocessable Content', which breaks three tests:
```
FAILED tests/renderers/httpdomain/test_render_response.py::test_render_response_content_custom[422]
FAILED tests/renderers/httpdomain/test_render_response_example.py::test_render_response_status_code[422]
FAILED tests/renderers/httpdomain/test_render_response_example.py::test_render_response_status_code_int[422]
```
 So these tests require some sort of conditional to change the expected output depending on the Python version.  I don't know how to do this "nicely", though.

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -10,10 +10,10 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Set up Python 3.11
+      - name: Set up Python 3.13
         uses: actions/setup-python@v4
         with:
-          python-version: '3.11'
+          python-version: '3.13'
 
       - name: Run pre-commit
         run: |
@@ -24,7 +24,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest]
-        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13']
 
     runs-on: ${{ matrix.os }}
     steps:
@@ -49,10 +49,10 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Set up Python 3.11
+      - name: Set up Python 3.13
         uses: actions/setup-python@v4
         with:
-          python-version: '3.11'
+          python-version: '3.13'
 
       - name: Run sphinx
         run: |
@@ -65,10 +65,10 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Set up Python 3.11
+      - name: Set up Python 3.13
         uses: actions/setup-python@v4
         with:
-          python-version: '3.11'
+          python-version: '3.13'
 
       - name: Prepare artifacts
         run: |

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -1,6 +1,6 @@
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.5.0
+    rev: v5.0.0
     hooks:
       - id: trailing-whitespace
       - id: end-of-file-fixer
@@ -14,7 +14,7 @@ repos:
       - id: debug-statements
 
   - repo: https://github.com/psf/black
-    rev: 23.10.0
+    rev: 24.10.0
     hooks:
       - id: black
         args: [--line-length=88, --target-version=py35]
@@ -46,6 +46,6 @@ repos:
           )
 
   - repo: https://github.com/pycqa/flake8
-    rev: 6.1.0
+    rev: 7.1.1
     hooks:
       - id: flake8

--- a/tests/renderers/httpdomain/test_render_response.py
+++ b/tests/renderers/httpdomain/test_render_response.py
@@ -1,9 +1,16 @@
 """OpenAPI spec renderer: render_response."""
 
+import sys
 import textwrap
 import pytest
 
 from sphinxcontrib.openapi import renderers
+
+
+if sys.version_info[:2] >= (3, 13):
+    http_422_description = "Unprocessable Content"
+else:
+    http_422_description = "Unprocessable Entity"
 
 
 def textify(generator):
@@ -218,7 +225,7 @@ def test_render_response_content_non_2xx(testrenderer, oas_fragment, status_code
         pytest.param("301", "Moved Permanently", id="301"),
         pytest.param("307", "Temporary Redirect", id="307"),
         pytest.param("401", "Unauthorized", id="401"),
-        pytest.param("422", "Unprocessable Entity", id="422"),
+        pytest.param("422", http_422_description, id="422"),
     ],
 )
 def test_render_response_content_custom(fakestate, oas_fragment, status_code, status):

--- a/tests/renderers/httpdomain/test_render_response_example.py
+++ b/tests/renderers/httpdomain/test_render_response_example.py
@@ -1,11 +1,18 @@
 """OpenAPI spec renderer: render_response_example."""
 
+import sys
 import textwrap
 
 import pytest
 import responses
 
 from sphinxcontrib.openapi import renderers
+
+
+if sys.version_info[:2] >= (3, 13):
+    http_422_description = "Unprocessable Content"
+else:
+    http_422_description = "Unprocessable Entity"
 
 
 def textify(generator):
@@ -499,7 +506,7 @@ def test_render_response_example_noop(testrenderer, oas_fragment):
     [
         pytest.param("201", "Created", id="201"),
         pytest.param("307", "Temporary Redirect", id="307"),
-        pytest.param("422", "Unprocessable Entity", id="422"),
+        pytest.param("422", http_422_description, id="422"),
     ],
 )
 def test_render_response_status_code(
@@ -577,7 +584,7 @@ def test_render_response_status_code_range(
     [
         pytest.param("201", "Created", id="201"),
         pytest.param("307", "Temporary Redirect", id="307"),
-        pytest.param("422", "Unprocessable Entity", id="422"),
+        pytest.param("422", http_422_description, id="422"),
     ],
 )
 def test_render_response_status_code_int(

